test/python/algorithms/optimizers/test_umda.py

```diff
@@ -0,0 +1,97 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Tests for the UMDA optimizer."""
+
+from test.python.algorithms import QiskitAlgorithmsTestCase
+from ddt import ddt
+
+import numpy as np
+from qiskit.algorithms.optimizers.umda import UMDA
+
+
+@ddt
+class TestUMDA(QiskitAlgorithmsTestCase):
+    """Tests for the UMDA optimizer."""
+
+    def test_truncation(self):
+        """Test if truncation is performed as expected"""
+        umda = UMDA(maxiter=100, size_gen=20, n_variables=10)
+
+        def objective_function(array):
+            return sum(array)
+
+        umda._check_generation(objective_function)
+        umda._truncation()
```

Comment:
In general a unit test is around the public API of the component under test. I would not expect unit tests to be calling methods that denoted private (or accessing any vars marked private - you don't have any but I made an earlier comment about all the instance vars being public). Rationale here is that we are testing public API and anything else is part of internal implementation that can be changed as long as public API is unaffected. Yes sometimes this makes things not so easy to get full test coverage.

Reply:
Okay, it is a pitty as a less exhaustive test can be done. However, I have changed the code so now it is tested in a different manner.